### DISCUSSION TITLE
Multicast command failing without reason

### DISCUSSION THREAD

**Discussion by dcmeglio** [zwave-js/zwave-js#7184] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
I was having an issue where I'd send an "on" command to 7 switches one after the other and was experiencing multiple failures. Someone suggested using multicast commands. Even with this some devices turn on, others do not respond. Feels like something is getting overloaded but 7 commands doesn't seem like a lot.

What did you expect to happen instead?
The switches should turn on.

  1. Using HA send a multicast command to 7 switch devices, only some will respond.

Anything else we should know?

### Software versions

Driver (node-zwave-js): 13.2.0

Z-Wave JS UI: 9.18.1

Home Assistant Z-Wave Integration: 2024.9.2

Home Assistant Z-Wave JS Addon: 3.11.0

ioBroker.zwave2 Adapter: I don't know what this is. 

If you are using something non-standard, tell us here: ...

### Device information

Does not appear to be a specific device but I was controlling these nodes 118, 80, 141, 26, 129, 58, 83

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwavejs_current.log](https://github.com/user-attachments/files/17091583/zwavejs_current.log)

**Comment by AlCalzone** [comment]
Which Z-Wave stick is that? Which firmware version is it on?

**Reply by dcmeglio** [reply]
SILabs SLUSB7000A UZB-7 running firmware 7.17.2 I have been considering updating the firmware but discussing with some people on the HASS Discord it sounded questionable if that's a good idea since the version I'm on is generally considered very stable, but I can do that if you suggest.

**Reply by dcmeglio** [reply]
FYI I also just updated Zwave JS just now since a new version is out, though I don't see anything in the release notes that seems related to my issue.

Driver (node-zwave-js): 13.3.1
Z-Wave JS UI: 9.20.0
Home Assistant Z-Wave JS Addon: 3.12.0

**Reply by AlCalzone** [reply]
Okay, so this is a difficult question to answer. 7.17.2 is indeed considered pretty stable, but there is something funky going on according to your log. `Fail` usually indicates that the controller was not able to transmit anything, but in this case it did. Some nodes respond, but apparently the controller no longer ACKs their commands. So obviously the firmware isn't perfect.

I think I saw the discussion and you were also having issues with non-multicast commands?

You can try upgrading, but you have to know there's no easy way back to 7.17.2.
The later 7.18.x versions are generally okay. If you want to go higher, go for 7.21.4 directly. Everything below that is a clusterfuck in one way or another.

**Reply by dcmeglio** [reply]
Yup that was me. Tell me what you think of this plan:
I have a new Zooz 800 stick coming tomorrow, I will upgrade that to 7.21.4, then I will do an NVM backup from my current SI Labs stick and restore on the Zooz (it should be ok that they're on different firmware to do the NVM backup/restore or no?)
That should rule out faulty hardware and firmware issues (I hope) and I get a nice new 800 controller too

**Reply by AlCalzone** [reply]
Sound good, except that you use Zooz's firmware 1.50, which is based on SDK 7.22.1 (exclusive to 800 series).
Doing the NVM backup/restore in Z-Wave JS UI makes sure the backup is converted to the correct format.

**Comment by dcmeglio** [comment]
Well the NVM backup/restore was incredibly smooth (kudos for that!) but
Unfortunately this did not resolve my issue. I'm now on an 800 series Zooz so I think we can rule out faulty hardware, and if it is firmware I'm now on a completely different firmware, Zooz 1.50

Note that same transmit fail at 2024-09-25T23:15:35.280Z

[zwavejs_current.log](https://github.com/user-attachments/files/17139525/zwavejs_current.log)

Anything else I can provide to figure out what's going on? Any work arounds you can think of? I can reproduce it pretty consistently

I ssh'ed into the addon container and have:
`tail -f zwavejs_current.log |grep "transmit status: Fail"`

I can get it to happen 100% of the time when I run my automation. I have a workaround of adding some delays but I'd love to find out if there's something I can do to track it down

**Reply by AlCalzone** [reply]
> figure out what's going on

Do you have anything you can turn into a [Zniffer](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/zniffer)? Preferably an UZB3 or the 800 series devkit? Your UZB-7 requires hooking up some pins on the board to an SWD debugger.

Otherwise you could try narrowing it down to specific end devices:
Take one half of the list, see if the issue reproduces.
Take the other half of the list, see if it reproduces there.
With each half, repeat the process until there are only 2 nodes left where it reproduces.

**Reply by AlCalzone** [reply]
Also, can you send me the NVM backup, so I can take a look at the routes? Maybe there's something messed up causing this.

**Reply by dcmeglio** [reply]
Closest thing I have is a ZWP-TBX (https://www.zwaveproducts.com/products/zwp-tbx-z-wave-toolbox) that I've added as a secondary controller in the past to troubleshoot zwave issues when I was on hubitat. Will that work?

 Happy to share my NVM backup, since it will have things like my encryption keys is there an email or somewhere I can send it instead of posting on github? Edit: Sent you the NVM backup via HA Discord

**Reply by AlCalzone** [reply]
Looks like that thing already has a Zniffer built in ("Packet Analyzer"). I'm not sure if it can export in a usable format, but that would be worth a shot.

I didn't spot anything weird in the NVM for the nodes you're communicating with, except that some have a hop in their LWR, so the initial multicast frame might not reach them all - they seem to react to the singlecast followup though.

What strikes me as odd, is that some nodes which you didn't send a command to are responding with their new state: Node 89 and 101. Do some of the targeted devices have an association to forward the command to others?

**Reply by dcmeglio** [reply]
Sorry I thought I grabbed enough of the log but I didn't :( the 101 and 89 are from a prior multicast command in the same automation.

I'll setup the packet analyzer this weekend and also grab better logs. I've used the packet analyzer before, it's pretty easy to use, I just join it to the network as a secondary controller. I'll see what format I can get data out in.

**Reply by dcmeglio** [reply]
Also is it worth switching back to single cast? It definitely seems to have something to do with the volume of commands. If I add delays in the automation it works perfectly. I can also share the automation config that is causing the issues if it helps.

**Reply by AlCalzone** [reply]
> Also is it worth switching back to single cast?

Multicast is mainly meant for reducing/eliminating the popcorn effect when controlling multiple devices. So if you were already having issues with singlecast, it may be worth looking at that first.

Oh, and just to be sure: Do you have the controller on a USB extension?
https://zwave-js.github.io/node-zwave-js/#/troubleshooting/first-steps?id=controller-placement

**Comment by Botched1** [comment]
Here for the "me too" that I'm having multicast issues. I do have a UZB3 flashed with sniffer firmware, but it is not hooked up right now. I can do that this weekend - is there any preferred way to get the sniffer logs in a format that is unable for you?

**Reply by AlCalzone** [reply]
Just use either the official (Windows) Zniffer application, or [Z-Wave JS UI](https://zwave-js.github.io/zwave-js-ui/#/usage/zniffer). Both can export in the same `.zlf` format.

**Reply by Botched1** [reply]
Here's a zniffer log of the error, captured in zwave js ui (native zniffer support was such a GREAT addition!).

[ZNIFFER.zip](https://github.com/user-attachments/files/17174699/ZNIFFER.zip)

Last error I saw was at 09:32am my local time (US central). Not sure how that translates to log time as the timestamps look really weird to me in the zlf.

@AlCalzone In case it is useful info, all 4 devices are of type:
![image](https://github.com/user-attachments/assets/58f59b9d-ac77-4af2-bc0f-87ccfee322ca)

I have been planning on replacing them with Zwave LR devices (have them on hand), but haven't yet in case this would be useful for troubleshooting.

Thanks,
Jason

**Reply by AlCalzone** [reply]
@Botched1 can you tell me which firmware (SDK version) is running on the controller and the end devices? You should be able to see the latter in the `Version` dropdown for each node in Z-Wave JS UI.

This Zniffer log looks like neither the end devices, nor the controller are acknowledging the routed singlecast frames following the multicast.

**Reply by AlCalzone** [reply]
Also, does this exact issue always happen with these devices? Or does it sometimes work? If so, a Zniffer log of how a successful case looks would be good aswell.

**Reply by Botched1** [reply]
> can you tell me which firmware (SDK version) is running on the controller and the end devices?

Controller = 7.22.1
End Devices = See below - I wasn't sure which number you needed.
![image](https://github.com/user-attachments/assets/096b6c21-9063-432a-8c8d-e443d0a44fde)


> Also, does this exact issue always happen with these devices? Or does it sometimes work? If so, a Zniffer log of how a successful case looks would be good as well.

It does sometimes work - actually works something like 8 out of 10 times. I'll get a zniffer log with that.

**Comment by dcmeglio** [comment]
> Multicast is mainly meant for reducing/eliminating the popcorn effect when controlling multiple devices. So if you were already having issues with singlecast, it may be worth looking at that first.

Yeah I am trying to eliminate popcorning too, but we can try with singlecast first. I'll make changes to my automation. I'll get the secondary controller setup this weekend too and see what I can capture.

> Oh, and just to be sure: Do you have the controller on a USB extension?

Yup, other than not as centrally located as would be ideal, all other guidelines are met. The USB extender has it at least 3-4ft from any other USB port or device and outside my server rack.

**Comment by dcmeglio** [comment]
I'm still working on this. I know it's just coincidence but I can't seem to reproduce it when I have the zniffer running. But unfortunately the issue is occurring every day in production use.

**Comment by dcmeglio** [comment]
I think I’m at the point where I need to just downgrade zwave js. Almost every zwave automation I have fails if it involves controlling 3 or more devices roughly 75% of the time. Not specific devices. Not consistent. Just random failure. Zwave has pretty much become unusable for me after working flawlessly for 2+ years. I haven’t been able to really capture much useful with the zniffer but I had 4 separate failures today alone. Other than zniffer logs is there anything else useful I can provide? The only workaround I’ve found is to add 5 second delays between every zwave command

**Comment by dcmeglio** [comment]
I rewrote all of my automations to be singlecast and I have had no issues. I was previously having issues but that was before I swapped out my controller. So I wonder if I'm having the same issue as @Botched1 now. I'll do some further troubleshooting

**Reply by dcmeglio** [reply]
I've been running singlecast automations for a week now and it's working perfectly. My original singlecast issues were when I was still on the 700 controller. I can only reproduce problems on 800 with multicast. I haven't found the secret to reproduce consistently, but I do think it is directly related to multicast.

**Reply by AlCalzone** [reply]
> I haven't found the secret to reproduce consistently, but I do think it is directly related to multicast.

Did you manage to get your traffic analyzer working? If you could provide a Zniffer trace which shows a working multicast and a failing multicast, plus a bit of time before each one, that might help.

**Reply by dcmeglio** [reply]
Yes, but the only export option it has is CSV. If you need some specific format for you to be able to analyze, just let me know what device I need to buy and I'll get it setup.

**Reply by AlCalzone** [reply]
Happy to look at one of those to see if it helps at all.

**Reply by dcmeglio** [reply]
[zwave_logs.zip](https://github.com/user-attachments/files/17335388/zwave_logs.zip)
zwave js logs during a success and failure and the network analyzer logs during the same time windows

In case it's helpful this is what I ran:

```
action: zwave_js.multicast_set_value
data:
  device_id:
# node 83
    - 060fca1a3da43092516b3f8a6468e541
# node 26
    - 47a51fb1c5eefe6acbda20d1430ecbb0
# node 118
    - 6db2e43b0dc4aebd47c85838db4bd09c
# node 80
    - dc32aa0c77f8996bf3a5693c336ece5c
 # node 141
    - b7e51fbfca1a052721858e5791bf4b87
  # node 58
    - 0462015716f9e0bf1ab11a448cb784ae
  command_class: "37"
  property: targetValue
  value: true
```

**Reply by dcmeglio** [reply]
Were you able to find anything in the logs?

**Reply by AlCalzone** [reply]
Sorry, your comment slipped through. The analyzer logs are a bit harder to decipher than Zniffer, but it confirms what I see in the Zniffer logs above. At least the controller sometimes stops responding to routed ACKs during multicast.

**Comment by AlCalzone** [comment]
I've opened a ticket with Silabs. Time to wait.

**Comment by AlCalzone** [comment]
@dcmeglio can you list the following for all end devices you were trying to reach via Multicast?
- Brand and Device name
- Firmware version
- If known: Z-Wave SDK version

**Reply by dcmeglio** [reply]
Well that's easy, they're all the same device! GE In Wall Receptacles. Note the one that is on a different FW than zwavejs says the FW is up to date, so not sure if it's a different underlying hardware version or something even though the model name is the same.

node 83
In-Wall Outlet, TR, 500S
FW: v5.26

node 26
In-Wall Outlet, TR, 500S
FW: v5.51
SDK: v6.82.0

node 118
In-Wall Outlet, TR, 500S
FW: v5.51
SDK: v6.82.0

node 80
In-Wall Outlet, TR, 500S
FW: v5.51
SDK: v6.82.0

node 141
In-Wall Outlet, TR, 500S
FW: v5.51
SDK: v6.82.0

node 58
In-Wall Outlet, TR, 500S
FW: v5.51
SDK: v6.82.0

**Reply by dcmeglio** [reply]
I take it no updates from SILabs?

**Reply by AlCalzone** [reply]
I actually missed a notification from them. Responded just now.

**Reply by dcmeglio** [reply]
Was there ever any response from SI Labs?

**Reply by AlCalzone** [reply]
They were able to reproduce something. I haven't heard if it's supposed to be fixed now. Will ask once I'm back from PTO.

**Comment by dcmeglio** [comment]
No worries, enjoy the PTO. I did a little digging:
https://www.silabs.com/documents/public/release-notes/SRN14930-7.23.0.0.pdf
Fix an issue where the Z-Wave stack is prevented from sending TX frames. The end device symptoms are a loss of packet transmission and no acknowledge emanating from the device. On the controller end, it materializes by a TX_COMPLETE_FAIL status answered back to the host. 

That sounds very much like my issue, but unfortunately it doesn't look like Zooz has created a firmware version for this yet :(

**Comment by ndoggac** [comment]
Seeing similar here.  On Aeotec 700 series, with FW: v7.21.6 and SDK: v7.21.6.   Which from the HASS website is a recommended version.

700 series:

prefer SDK versions 7.17.2 to 7.18.x or 7.21.6 and newer
SDK versions 7.19.x are okay
avoid SDK versions before 7.17.2
avoid SDK versions 7.20 to 7.21.5


The device that "failed to acknowledge" is random, and NOT device or manufacturer specific.  No security on any devices, mix of non-plus, +1, and +2 devices.

I can run the same z-wave multicast script (or section of script) 20 times, and it will both fail & succeed randomly.   When it fails it's a different device nearly every time.  Happens on both command class 37 and 38 targets (binary & dimmer).  Happens whether you utilize device id or entity id, 

manifests in logs as:
`2025-07-16 10:35:47.244 ERROR Z-WAVE-SERVER: The node did not acknowledge the command (ZW0204)
ZWaveError: The node did not acknowledge the command (ZW0204)
    at Driver.sendMessage (/opt/node_modules/zwave-js/src/lib/driver/Driver.ts:7057:23)
    at Driver.sendCommandInternal (/opt/node_modules/zwave-js/src/lib/driver/Driver.ts:7253:28)
    at Driver.sendCommand (/opt/node_modules/zwave-js/src/lib/driver/Driver.ts:7371:29)
    at Proxy.set (/opt/node_modules/@zwave-js/cc/src/cc/MultilevelSwitchCC.ts:246:20)
    at Proxy.set (/opt/node_modules/@zwave-js/cc/src/cc/MultilevelSwitchCC._validateArgs.ts:13:30)
    at Proxy.<anonymous> (/opt/node_modules/@zwave-js/cc/src/cc/MultilevelSwitchCC.ts:323:17)
    at VirtualNode.setValue (/opt/node_modules/zwave-js/src/lib/node/VirtualNode.ts:169:39)
    at MulticastGroupMessageHandler.handle (/opt/node_modules/@zwave-js/server/dist-esm/lib/multicast_group/message_handler.js:16:50)
    at Client.receiveMessage (/opt/node_modules/@zwave-js/server/dist-esm/lib/server.js:134:100)
    at WebSocket.<anonymous> (/opt/node_modules/@zwave-js/server/dist-esm/lib/server.js:55:45)`

**Comment by ndoggac** [comment]
I'm still having this problem even after switching to the new Nabu Casa Zwave controller (ZWA-2, NC-ZWA-9734 on FW v1.1 & SDK v7.23.1).  I think perhaps I've tracked this down to a specific device, or set of devices under different manufacturers.  The issue may only manifest when you have more than 2 of these devices in the same multicast command for binary (37) switches, but 1 of these devices in a dimmer multicast command (38) seems to be enough?

Can everyone check if they have the following devices in their Zwave mesh, and if you eliminate them from the multicast commands, that things start to work?  If so, start adding them back in and see if it always works with 1 or 2, but not 3 of these devices in a single multicast command.

UltraPro	In-Wall Paddle Switch, QFSW, 500S	39348 / 54890 / 54891 / ZW4008
FW: v5.54 and SDK v6.81.3

or

UltraPro	In-Wall Paddle Dimmer, QFSW, 500S	39351 / ZW3010
FW: v5.54 and SDK: v6.81.3

or

Honeywell	In-Wall Paddle Switch, QFSW, 500S	39348 / 39455 / ZW4008
FW: v5.53 and SDK v6.81.3

I'm not sure if these models would appear under other manufacturers as well (GE, Enbrighten, etc.) with the same firmware?  I also have 300S (GE Dimmer/Switch/Outlet) & 700S (Enbrighten Switch) models as well, but they don't seem to be causing problems.

From looking at the logs, the 500S model is nearly always the culprit.  With two of them in a binary (37) multicast command, I can run the script every 5 seconds and it won't fail.  As soon as I add a third device, it starts failing randomly.  For dimmers multicast commands (38) one 500S devices is enough to cause a failure.  When a 500S is not the culprit in the logs, I believe it's because the return message from another device is routing via a 500S back to the controller, as removing the 500S from that command eliminates all problems.

Edit:  I see now above that 500S models were already called out back in October, but still no firmware/sdk fix?  Any updates on this? These are fairly modern Zwave+ version 1 devices.  I have 17 of these devices - would love to get this fixed!!  SDK v7.23.1 on the controller does not fix this issue.  This is pretty concerning, considering it's across a wide range of device types, and manufacturers, and is a long-standing bug (FW: 5.26 - 5.54, or SDK 6.81.3 - 7.23.1).

**Reply by cperuffo3** [reply]
Hilariously enough, I actually do have an UltraPro 500S in-network as well. Paddle Switch, FW v5.54, SDK v6.81.3 -- 

I'll note however as of the latest chipset firmware update on my controller, the issue has drastically been reduced, but definitely not eliminated. 

I'll attempt to replace this guy and see how the network handles it

**Reply by ndoggac** [reply]
Thanks for confirming.  For the binary switches sending 37 multicast, I think previous fixes have improved (not eliminated like you said) the situation for 1-2 500S devices within the command.  As soon you hit 3 500S devices within the command it's completely broken again.  

For the 500S dimmer switches sending 38 multicast, all it takes is a single 500S device within the command to break it.  

In either situation, logs might show an alternative device as the culprit, but I believe it's because the 500S is dropping that device's routed response through the mesh, or something similar.  Removing the 500S from the command completely eliminates all failures, but removing the alt device (with the 500S in) does not.

**Reply by AlCalzone** [reply]
Does anyone of you have a Zniffer, or a controller you can [turn into one](https://zwave-js.github.io/zwave-js/#/troubleshooting/zniffer)?

Without seeing what actually happens, we can't even make a bug report with Silabs.

**Reply by ndoggac** [reply]
If the hardware will work, I have my old Aeotec 700 series USB (switched to Nabu Casa ZWA-2).  
I could give it a go.  
When I purchased the Aeotec it's a 700 series, but before I removed it from HASS it showed up as a 700/800 series label.  Should I utilize Gecko or Simplicity for this hardware?

Will this work as a zniffer integrated into my existing HASS, or should I setup it up in a separate instance?

**Reply by AlCalzone** [reply]
I'm not sure that one will work. Also from my experience it doesn't have a good antenna so you might miss traffic.

The controllers linked in the docs or the xG28 Explorer Kit should work. Or a second ZWA-2 for that matter.

**Reply by ndoggac** [reply]
> I'll attempt to replace this guy and see how the network handles it

cperuffo3 any chance you eliminated the 500S device from your network and retried multicast?  did it help?

**Comment by ndoggac** [comment]
I've narrowed down behavior of 500S devices a bit more.  I have 18 500S devices total, 9 Honeywell binary, 8 UltraPro binary, and 1 UltraPro Dimmer.  It seems the UltraPro binary switches are even worse than the Honeywell binary switches.  

There's likely an issue with Zooz Zen22 dimmers as well.  Any chance HASS folks could reach out to Zooz on this one?  If they could narrow it down, it might point to a fix for the others.

I've setup a multicast test script where I can enable and disable different functions, and run them on a 3 second timer to brute force this.  Each of these tests were tested on 3 subset device combinations where possible to ensure it wasn't any one device misbehaving, and the behavior was linked to device type.

Multicast 37 off 1 Honeywell switch - runs every 3 seconds with no issues
Multicast 37 off 2 Honeywell switches - runs with rare failures
Multicast 37 off 3+ Honeywell switches - runs sporadically, but often breaks

Multicast 37 off 1 UltraPro switch - runs every 3 seconds with no issues
Multicast 37 off 2 UltraPro switches - runs sporadically, but often breaks
Multicast 37 off 3+ UltraPro switches - runs sporadically, but often breaks

Multicast 37 off 2 Honeywell & 1 UltraPro - runs with rare failures
Increase either of the 2 device numbers above, and runs sporadically, but often breaks

Multicast 37 off to 7 non-500S devices I have in addition - runs every 3 seconds with no issues

Multicast 38 off to 1 UltraPro 500S dimmer - runs every 3 seconds with no issues (only 500S dimmer I have)

Multicast 38 off to 7 non-500S Dimmers - runs sporadically, but often breaks
what else is causing issues?
Multicast 38 off to 4 non-500S GE Dimmers - runs every 3 seconds with no issues
Multicast 38 off to 2 Zooz Z22 Dimmers - runs with rare failures
Multicast 38 off to 1 Zooz Z72 Dimmer - runs every 3 seconds with no issues

Multicast 38 off to 5 (4 GE & 1 Zen22 Zooz) - runs sporadically, but often breaks
Multicast 38 off to 6 (4 GE & 2 Zen22 Zooz) - runs sporadically, but often breaks

Multicast 38 off to 5 (4 GE & 1 Zen72 Zooz) - runs every 3 seconds with no issues

Multicast 38 off to 4 GE, 1 UltraPro 500S - runs with rare failures
Multicast 38 off to 4 GE, 1 Zooz Z72, 1 UltraPro 500S - runs with rare failures

I'm performing some additional testing with simple light.turn_on/off commands to see if these 500S devices perform poorly in the same manner.  Large turn_off scripts turning off ~25 devices can regularly take 22 seconds to complete.  I'm trying to sort through why now.  I will report back.

**Comment by mcondren** [comment]
I am still experiencing this with FW: v1.60 and SDK: v7.23.2.  It happens about once per day.  Hit the control for multicast, all lights go out but one.  Wait a bit...nothing...hit the button again and the last one goes off.  I tried to keep the debug window open but I'm not seeing any erorrs.
